### PR TITLE
Main recipe card

### PR DESCRIPTION
**Pull request by GTZL1**

### CODE REVIEW

File: src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt [hunk 1]
```diff
@@ -4,31 +4,45 @@ import androidx.compose.foundation.Icon
...
     BottomAppBar(backgroundColor = Color.White)
     {
         Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceEvenly)
         {
-            IconButton(onClick = {}) {
-                Icon(vectorResource(id = R.drawable.ic_navigate_before_black_18dp))
+            Button(modifier = Modifier.size(45.dp), onClick = onReturn, shape = CircleShape, backgroundColor = Color(
+                197, 180, 246)) {
+                // TODO find a way to display these buttons in white
+                Icon(vectorResource(id = R.drawable.ic_navigate_before_white_18dp))
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt]: Use figma icon

File: src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt [hunk 2]
```diff
@@ -4,31 +4,45 @@ import androidx.compose.foundation.Icon
...
+    onRecipeClick: () -> Unit
+) {
     BottomAppBar(backgroundColor = Color.White)
     {
         Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceEvenly)
         {
-            IconButton(onClick = {}) {
-                Icon(vectorResource(id = R.drawable.ic_navigate_before_black_18dp))
+            Button(modifier = Modifier.size(45.dp), onClick = onReturn, shape = CircleShape, backgroundColor = Color(
+                197, 180, 246)) {
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt]: Extract to resources if not already in our theme

File: src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt [hunk 3]
```diff
@@ -1,27 +1,43 @@
...
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import tupperdate.android.R
 
 @Composable
-fun mainTopBar() {
+fun mainTopBar(
+    onChatClick: ()->Unit,
+    onProfileClick: ()->Unit,
+    modifier: Modifier=Modifier
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt]: `modifier` is never passed to `TopAppBar` rendering it useless

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 4]
```diff
@@ -1,98 +1,111 @@
...
-import tupperdate.api.AuthenticationApi
+import tupperdate.api.RecipeApi
+import tupperdate.api.api
+import kotlin.math.absoluteValue
 
 @Composable
+/*
+@ Guy-Laurent : I am forced to do only one function because it is the only way
+to call like() and dislike() without the weird RealRecipeApi import.
+ */
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt:26]: Bruh...

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 5]
```diff
@@ -1,98 +1,111 @@
...
 fun Home(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile?,
-    modifier: Modifier = Modifier,
+    recipeApi: RecipeApi,
+    onChatClick: () -> Unit,
+    onProfileClick: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt:33]: We should have a second function that isn't reliant on the api.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 6]
```diff
@@ -1,98 +1,111 @@
...
-    }
+    )
 }
 
 @Composable
-private fun HomeConnected(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun defaultContent(
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: Should start with an uppercase letter.
Also, the name isn't very explicit xD

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 7]
```diff
@@ -1,98 +1,111 @@
...
+    presentRecipe: RecipeApi.Recipe,
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    modifier: Modifier = Modifier
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: Modifier is never passed to the `Row`

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 8]
```diff
@@ -1,98 +1,111 @@
...
+                        }
+                    }, onDragStopped = { if (pos.absoluteValue < swipeMargin) setPos(0) })
+            )
+            {
+                if (pos.absoluteValue > swipeMargin) {
+                    if (pos > 0) {
+                        onLike
+                    } else if (pos < 0) {
+                        onDislike
+                    }
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: `onLike` and `onDislike` are methods... You should have parentheses

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 9]
```diff
@@ -1,98 +1,111 @@
...
-    onButtonClick: () -> Unit,
-    modifier: Modifier,
-) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there.")
-        Button(onClick = onButtonClick) {
-            Text("Go to onboarding")
+    // TODO make a call to front element of recipeApi
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt:34]: Probably use something like `val stack by recipeApi.stack().collectAsState(emptyList())`

**Review by glsubri — CHANGES_REQUESTED** [review]
Also, please apply android studio's formatting...

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt (hunk 5 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Home.kt:33]: I explained why

File: src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt (hunk 2 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt]: it's done

File: src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt (hunk 3 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt]: Yeah I removed

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt (hunk 6 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Home.kt]: changed

**Pull request closed without merging**